src/FramePlayer.cpp

```diff
@@ -0,0 +1,102 @@
+/* Modified from https://github.com/facebookresearch/vrs
+ *
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iostream>
+#include <memory>
+#include <sstream>
+
+#include <vrs/DataLayout.h>
+#include <vrs/utils/PixelFrame.h>
+
+#include "FramePlayer.h"
+#include "utils.h"
+
+namespace rerun_vrs {
+
+    struct FrameNumberDataLayout : public vrs::AutoDataLayout {
+        vrs::DataPieceValue<uint64_t> frameNumber{"frame_number"};
+
+        vrs::AutoDataLayoutEnd endLayout;
+    };
+
+    FramePlayer::FramePlayer(vrs::StreamId id, std::shared_ptr<rerun::RecordingStream> rec)
```

Comment:
```suggestion
    FramePlayer::FramePlayer(vrs::StreamId id, std::shared_ptr<const rerun::RecordingStream> rec)
```

Reply:
here and everywhere, to make it clear it is thread-safe

Reply:
Changed.